### DISCUSSION TITLE
Show a specific icon if the menu is open/closed

### DISCUSSION THREAD

**Discussion by JulienLecoq** [ng-primitives/ng-primitives#69] [Q&A] [answered]
Is it possible to show a chevron icon down/up if the menu is open or closed using the https://angularprimitives.com/primitives/menu primitive?

**Comment by ashley-hunter** [comment, marked as answer]
Hey! Absolutely, you can show and hide icons based on the presence of the `[data-open]` attribute on the trigger element:

https://stackblitz.com/edit/angular-aeqwvf?file=src%2Fmain.ts

If you have a different scenario let me know and I can investigate further. Thanks

**Comment by JulienLecoq** [comment]
Oh I'm dumb, I didn't think of doing it that way with css.. I found a solution using Typescript and a mutation observer on the button that listen to changes on the `data-open` attribute and that change the value of a signal but that's a lot more code than just doing what you did lol. Thanks a lot for the quick and good response :)

**Comment by JulienLecoq** [comment]
I'm using the lib since the beginning of this afternoon and it's so far so good!

**Reply by ashley-hunter** [reply]
No problem! No, it's not the most obvious thing. I think we should have a guide on how to style things using data attributes, as this is a pattern that applies to many primitives, but arguably in this case we could also have an output where you can be informed programmatically about state changes.

Thank you for positive feedback! If you have any suggestions be sure to get in touch 😊
